tests/raftstore/cluster.rs

```diff
@@ -146,19 +146,42 @@ impl<T: Simulator> Cluster<T> {
         self.sim.rl().call_command(request, timeout)
     }
 
+    fn naive_call_command_on_leader(&mut self,
+                                    region_id: u64,
+                                    mut request: RaftCmdRequest,
+                                    timeout: Duration)
+                                    -> Result<RaftCmdResponse> {
+        if let Some(leader) = self.leader_of_region(region_id) {
+            request.mut_header().set_peer(leader);
+            return self.call_command(request, timeout);
+        }
+        Err(box_err!("can't get leader of region"))
+    }
+
     pub fn call_command_on_leader(&mut self,
                                   region_id: u64,
-                                  mut request: RaftCmdRequest,
+                                  request: RaftCmdRequest,
                                   timeout: Duration)
                                   -> Result<RaftCmdResponse> {
-        for _ in 0..200 {
-            if let Some(leader) = self.leader_of_region(region_id) {
-                request.mut_header().set_peer(leader);
-                return self.call_command(request, timeout);
+        let mut retry_cnt = 0;
+        loop {
```

Comment:
let mut result: Result<RaftCmdResponse>;
        for _ in 0..retry_cnt {
                result = ...;
                if let Ok(_) = result {
                        coutinue
                }
        }
        result

compiler would complain "use of partially moved value" of result